### PR TITLE
feat: localup shell adds generate sp.info and db.info function

### PR DESCRIPTION
**Pull request by sysvm**
### Description

localup shell adds generate sp.info and db.info function

### Rationale

Easier deployment of local SP environments.

### Example

`bash deployment/localup/localup.sh --generate sp.json db_username db_password db_address`

### Changes

N/A

### CODE REVIEW

**Review by RenRick — APPROVED** [review]

File: deployment/localup/localup.sh [hunk 1]
```diff
@@ -10,56 +10,77 @@ sp_bin=${workspace}/../../build/${sp_bin_name}
...
+  fi
+  rm -rf ${workspace}/${SP_DEPLOY_DIR}
   bash ${workspace}/../../build.sh
   mkdir -p ${workspace}/${SP_DEPLOY_DIR}
 
+  sp_json_file=$1
+  user=$2
+  password=$3
+  address=$4
+  port=9033
```
Comment (xibao-nr) [inline deployment/localup/localup.sh:41]: Move `port=9033` to env.sh?

**Review by xibao-nr — APPROVED** [review]

**Review by will-2012 — APPROVED** [review]

File: deployment/localup/localup.sh (hunk 1 above)
Comment (sysvm) [reply deployment/localup/localup.sh]: fixed

File: deployment/localup/localup.sh [hunk 2]
```diff
@@ -10,56 +10,75 @@ sp_bin=${workspace}/../../build/${sp_bin_name}
...
+    op_address=$(jq -r ".sp${i}.OperatorAddress" ${sp_json_file})
+    echo "OPERATOR_ADDRESS=\"${op_address}\"" >> sp.info
+    opk=$(jq -r ".sp${i}.OperatorPrivateKey" ${sp_json_file})
+    echo "OPERATOR_PRIVATE_KEY=\"${opk}\"" >> sp.info
+    fpk=$(jq -r ".sp${i}.FundingPrivateKey" ${sp_json_file})
+    echo "FUNDING_PRIVATE_KEY=\"${fpk}\"" >> sp.info
+    spk=$(jq -r ".sp${i}.SealPrivateKey" ${sp_json_file})
+    echo "SEAL_PRIVATE_KEY=\"${spk}\"" >> sp.info
+    apk=$(jq -r ".sp${i}.ApprovalPrivateKey" ${sp_json_file})
+    echo "APPROVAL_PRIVATE_KEY=\"${apk}\"" >> sp.info
```
Comment (Fynnss) [inline deployment/localup/localup.sh:62]: Missing GC private key?

Comment (sysvm) [reply deployment/localup/localup.sh:62]: added

File: deployment/localup/localup.sh [hunk 3]
```diff
@@ -191,11 +187,37 @@ print_work_dir() {
...
+  done
+}
+
+#############
+# reset sps #
+#############
+reset_sp() {
+  stop_sp
+  reset_db
+  reset_store
```
Comment (Fynnss) [inline deployment/localup/localup.sh:208]: What's the difference of `reset_db` and `reset_store`?

Comment (Fynnss) [reply deployment/localup/localup.sh:208]: how about `reset_sqldb` and `reset_piece_store`?

Comment (sysvm) [reply deployment/localup/localup.sh:208]: fixed

**Review by Fynnss — APPROVED** [review]

**Pull request merged**